source/whats-new.txt

```diff
@@ -22,11 +22,19 @@ Overview
 
 Learn what's new in:
 
+* :ref:`Version 4.1 <version-4.1>`
 * :ref:`Version 4.0 <version-4.0>`
 * :ref:`Version 3.11 <version-3.11>`
 * :ref:`Version 3.10 <version-3.10>`
 * :ref:`Version 3.9 <version-3.9>`
 
+.. _version-4.1:
+
+What's New in 4.1
+-----------------
+
+- Adds support for C++23.
```

Comment:
I do not think this entry is necessary. This is not a "new" feature specific to v4.1. I believe it is enough for documentation to reflect existing support (since the 3.9.0 release which addressed C++20 compatibility issues). The original ticket which prompted this doc update did little more than extend our compilation test coverage to include C++23 ([CXX-3187](https://jira.mongodb.org/browse/CXX-3187)).

Reply:
I came to the same conclusion while writing this and did not intend to include it in the PR. Apologies.